liquidity-mining/src/tests/tests.rs

```diff
@@ -1522,89 +1611,278 @@ fn update_yield_farm_should_work() {
             );
 
             if current_period != yield_farm_updated_at && !yield_farm_total_valued_shares.is_zero() {
                 frame_system::Pallet::<Test>::assert_has_event(mock::Event::LiquidityMining(
                     Event::YieldFarmAccRPVSUpdated {
-                        global_farm_id: global_farm.id,
-                        yield_farm_id: *yield_farm_id,
-                        accumulated_rpvs: FixedU128::from_inner(*expected_yield_farm_accumulated_rpvs),
-                        total_valued_shares: yield_farm.total_valued_shares,
+                        global_farm_id: global_farm_0.id,
+                        yield_farm_id: yield_farm_0.id,
+                        accumulated_rpvs: FixedU128::from(expected_yield_farm_accumulated_rpvs),
+                        total_valued_shares: yield_farm_0.total_valued_shares,
                     },
                 ));
             }
         });
     }
 }
 
+#[test]
+fn sync_yield_farm_should_now_update_when_yield_farm_is_not_acitve() {
+    let global_farm_0 = GlobalFarmData {
+        id: 1,
+        owner: ALICE,
+        updated_at: 1000,
+        total_shares_z: 1_000_000_u128,
+        accumulated_rpz: FixedU128::from(10),
+        reward_currency: BSX,
+        pending_rewards: 1_000_000 * ONE,
+        accumulated_paid_rewards: 1_000_000 * ONE,
+        yield_per_period: Perquintill::from_float(0.5_f64),
+        planned_yielding_periods: 1_000,
+        blocks_per_period: 1,
+        max_reward_per_period: 10_000 * ONE,
+        incentivized_asset: BSX,
+        min_deposit: crate::MIN_DEPOSIT,
+        live_yield_farms_count: Default::default(),
+        total_yield_farms_count: Default::default(),
+        price_adjustment: FixedU128::one(),
+        state: FarmState::Active,
+    };
+
+    let yield_farm_0 = YieldFarmData {
+        id: 2,
+        updated_at: 50,
+        total_shares: 200_u128,
+        total_valued_shares: 10_000 * ONE,
+        accumulated_rpvs: FixedU128::from(3),
+        accumulated_rpz: FixedU128::from(4),
+        loyalty_curve: None,
+        multiplier: FixedU128::from(2_u128),
+        state: FarmState::Active,
+        entries_count: 0,
+        left_to_distribute: 0,
+        total_stopped: 0,
+        _phantom: PhantomData::default(),
+    };
+
+    let mut global_farm = global_farm_0.clone();
+    let mut stopped_yield_farm = yield_farm_0.clone();
+    stopped_yield_farm.state = FarmState::Stopped;
+    let mut terminated_yield_farm = yield_farm_0.clone();
+    terminated_yield_farm.state = FarmState::Terminated;
+
+    new_test_ext().execute_with(|| {
+        let current_period = yield_farm_0.updated_at + global_farm_0.updated_at;
+        //Stopped yield-farm
+        assert_ok!(with_transaction(|| {
+            TransactionOutcome::Commit(LiquidityMining::sync_yield_farm(
+                &mut stopped_yield_farm,
+                &mut global_farm,
+                current_period,
+            ))
+        }));
```

Comment:
```suggestion
        assert_ok!(commit(LiquidityMining::sync_yield_farm(
                &mut stopped_yield_farm,
                &mut global_farm,
                current_period,
            )));
```
might be nice to have a test util function with usage like above?

or even a macro `assert_transact_ok` or similar